docs/nomt_specification.md

```diff
@@ -0,0 +1,176 @@
+
+Each sextet defines which page table to go to next, starting from the root page.
+
+If d is a divisor of the key length, then there are `256/d` Sextets, each 'd'-bits long; otherwise, Sextets are variable length. The first `⎣256/d⎦` Sextets are 'd'-bits long, while the last one is `256 - (⎣256/d⎦ * d)` bits long. (TODO: Does this make sense?)
+
+Each page will be assigned a unique identifier (**PageId**) created by combining the sextets required to reach that page. As mentioned earlier, multiple sextets forming a path in the page tree allow for traversal of the page tree.
+
+###  From KeyPath to Pages
+
+Given a Node Key, we can treat it as an array of bits. Starting from zero, each sub-array of a length that is a multiple of `d` is considered a PageId. If `d` is not a divisor of the key length, then all the mentioned sub-arrays are PageIds, and the entire array is also a PageId.
+
+Example with key length equals to 16 and `d` equals 6:
+
+KeyPath = |000001|000010|00011
+
+PageId_1 = 000001
+
+PageId_2 = 000001000010
+
+PageId_3 = 00000100001000011
+
+And here's how we can deterministically get the value and all the parent nodes in Merkle form from the pages identified by the PageIds extracted from the KeyPath.
+
+### Rootless sub-binary tree in Page
+ 
+A Page is just an array of bytes that will contain a rootless sub-binary tree. 
+
+#### Node
+
+Each node will consist of 32 bytes, with the most significant bit (msb) determining whether the node is an internal node (0 as msb) or a leaf (1 as msb). A node with 32 bytes of all zeros is a special node called the empty node.
+
+Internal nodes will store the hash of the concatenation of the hashes of its children. The children can be internal nodes, in which case their hashes are used; they could also be empty (using 32 bytes of zeros), or they could be leaves, in which case the hash of the value is used.
+
+#### Page
+
+Each node in the rootless sub-binary tree is assigned a position. The root is assigned position 1, and at each subsequent level, the identifiers increase from left to right. For example:
+
+![nomt identifier rule](./images/nomt_number_rule.png)
+
+Each node at position `p` in the tree will be stored in the array at index `p - 1`. Since it is rootless, `p` will always be greater than 1. The children of a node at position `p` will be at positions `p * 2` and `(p * 2) + 1`, corresponding to the left and right child nodes, respectively.
+
+All nodes that are not leaves or internal nodes are empty when viewed from a page perspective. However, it is possible to access only few of them respecting the constraints outlined in the previous section.
+
+### Storing page on disk (TODO: maybe remove this being not relevant with the first implementation)
```

Comment:
Better skip such implementation details for the moment. Broad strokes are in the data format - we can document algorithms and caching strategies once direction is clearer.